### PR TITLE
Feature/database

### PR DESCRIPTION
**Pull request by Boukhari-Mohamed**
first draft of the db implementation

### CODE REVIEW

File: app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/database/firebase/FirebaseDB.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    private static final String TAG = "FirebaseDB";
+    private static FirebaseDB instance;
+
+
+    private static FirebaseFirestore database;
+
+
+    public FirebaseDB() {
+        //database = FirebaseFirestore.getInstance();
+        //FirebaseAuth.getInstance();
```
Comment (Roilek) [inline app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/database/firebase/FirebaseDB.java:27]: Why is there only commented out code?

File: app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/database/firebase/FirebaseDB.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    public Task<String> add(String collectionPath, Map<String, Object> data) {
+        return database.collection(collectionPath)
+                .add(data).onSuccessTask((documentReference ->
+                        Tasks.forResult(documentReference.getId())
+                ));
+    }
+
+    @Override
+    public Task<DBDataCollection> fetchAll(String collectionPath) {
+        return null;
```
Comment (Roilek) [inline app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/database/firebase/FirebaseDB.java:67]: Is it planned to be implemented ?

File: app/src/test/java/com/epfl/neighborfood/neighborfoodandroid/database/DatabaseFactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                return null;
+            }
+
+            @Override
+            public Task<DBDataCollection> fetchAll(String collectionPath) {
+                return null;
+            }
+        };
+
+        //DatabaseFactory.setDependency(dep);
```
Comment (Roilek) [inline app/src/test/java/com/epfl/neighborfood/neighborfoodandroid/database/DatabaseFactoryTest.java:47]: Why is there commented out code ?

**Review by Roilek — COMMENTED** [review]
There is a need to implement some (very basic?) tests in order to get the coverage.
Some code is still here but commented out, should it be removed?
Code pretty clear overall, mainly short methods easy to understand.

File: app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/database/firebase/FirebaseDB.java (hunk 2 above)
Comment (Boukhari-Mohamed) [reply app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/database/firebase/FirebaseDB.java:67]: Yes it is planned to be implemented

File: app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/database/firebase/FirebaseDB.java (hunk 1 above)
Comment (Boukhari-Mohamed) [reply app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/database/firebase/FirebaseDB.java:27]: These lines should be deleted, I was testing a problem with FireBase. I will fix them shortly

File: app/src/test/java/com/epfl/neighborfood/neighborfoodandroid/database/DatabaseFactoryTest.java (hunk 3 above)
Comment (Boukhari-Mohamed) [reply app/src/test/java/com/epfl/neighborfood/neighborfoodandroid/database/DatabaseFactoryTest.java:47]: Whenever Firebase is called within a unit test, the tests crashes. I am still looking for an explanation or a workaround , but this won't be done this sprint

File: app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/database/firebase/FirebaseDB.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+
+    @Override
+    public Task<DBData> fetch(String collectionPath, String documentPath) {
+        return database.collection(collectionPath).document(documentPath)
+                .get().onSuccessTask((documentSnapshot ->
+                        Tasks.forResult(new FirebaseData(documentSnapshot))
+                ));
+    }
```
Comment (raedbm) [inline app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/database/firebase/FirebaseDB.java:44]: Can you explain why you are using paths for the database queries

Comment (raedbm) [inline app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/database/firebase/FirebaseDB.java:44]: Can you explain why you are using paths for the database queries

**Review by raedbm — APPROVED** [review]
This pull review is well done , code is clean and commented, the functions for the database queries are pretty intuitive and not hard to use later, maybe you should add more tests as elior suggested to have more code coverage.

File: app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/database/firebase/FirebaseDB.java (hunk 4 above)
Comment (Boukhari-Mohamed) [reply app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/database/firebase/FirebaseDB.java:44]: the paths will allow us to have different collections in our db and enables faster access to documents in some cases

Comment (Boukhari-Mohamed) [reply app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/database/firebase/FirebaseDB.java:44]: Response above

**Review by badismach — APPROVED** [review]
Code is well written and commented. Not much to add to Raed and Elior's comments, so I will approve to merge. Maybe you could use more expressive names for functions/interfaces.?

**Comment by Roilek** [comment]
We have to increase the coverage but the implementation of this PR for the database looks good.

**Pull request merged**